### PR TITLE
Update NWB Basics tutorial

### PR DESCRIPTION
**Pull request by weiglszonja**
Extends the NWB File Basics tutorial with:

- merging the existing PR #1402 related changes with this branch
- more data types covered (e.g. Subject, SpatialSeries, Position)
- added UML diagrams to help understanding of inheritance, extensions
- merge existing tutorial sections with the beginning sections of [nwb_tutorial](https://github.com/NeurodataWithoutBorders/nwb_tutorial/blob/master/HCK11/ophys_tutorial.ipynb)
- removed the Units part (better to have it in the ecephys tutorial #1391)

## Checklist

- [x] Did you update CHANGELOG.md with your changes?
- [x] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [x] Have you ensured the PR clearly describes the problem and the solution?
- [x] Is your contribution compliant with our coding style? This can be checked running `flake8` from the source directory.
- [x] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/NeurodataWithoutBorders/pynwb/pulls) for the same change?
- [x] Have you included the relevant issue number using "Fix #XXX" notation where XXX is the issue number? By including "Fix #XXX" you allow GitHub to close issue #XXX when the PR is merged.

### CODE REVIEW

**Comment by oruebel** [comment]
Just a heads up. I also made some changes to ``docs/gallery/general/file.py`` to merge the content of ``docs/source/overview_nwbfile.rst`` with the tutorial in the following PR https://github.com/NeurodataWithoutBorders/pynwb/pull/1402

**Comment by weiglszonja** [comment]
@oruebel Thanks for the heads-up! I'll close this PR and create a new tutorial page focusing on adding behavior data that is not covered yet in the tutorials (Subject, SpatialSeries, Position, etc.) or I can merge your changes here; What do you think?

**Comment by oruebel** [comment]
> create a new tutorial page focusing on adding behavior data

@weiglszonja I agree, adding a tutorial focused on behavioral data will be useful. Depending on the scope that tutorial may fit well in the  "Domain-specific tutorials" section.

BTW, if there are any changes from this PR that you think should remain part of the "NWB File basics" tutorial instead, then please feel free to make those changes in this or a new PR. I mainly wanted to give you a heads up about the other PR in case of merge conflicts. However, since the other PR mainly expands on the intro part of the "NWB File basics" tutorial, I don't think merging the changes should be too complicated.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,15 @@
 - Fix bad error check in ``IntracellularRecordingsTable.add_recording`` when adding ``IZeroClampSeries``. @oruebel (#1410)
 -  Skip ros3 tests if internet access or the ros3 driver are not available. @oruebel (#1414)
 
+### Tutorial enhancements:
+- Updated the general tutorial to add documentation about the ``Images`` type. @bendichter (#1353)
```
Comment (bendichter) [inline CHANGELOG.md]: Tis info is repeated

File: docs/gallery/general/file.py [hunk 2]
```diff
@@ -1,76 +1,381 @@
...
-create_date = datetime(2017, 4, 15, 12, tzinfo=tzlocal())
+nwbfile = NWBFile(
+    session_description='Mouse exploring an open field',     # required
+    identifier='Mouse5_Day3',                                # required
+    session_start_time=session_start_time,                   # required
+    session_id='session_1234',                               # optional
+    experimenter='My Name',                                  # optional
+    lab='My Lab Name',                                       # optional
+    institution='University of My Institution',              # optional
+    related_publications='DOI:10.1016/j.neuron.2016.12.011'  # optional
```
Comment (bendichter) [inline docs/gallery/general/file.py]:
```suggestion
    related_publications='DOI:10.1016/j.neuron.2016.12.011', # optional
```

File: docs/gallery/general/file.py [hunk 3]
```diff
@@ -1,76 +1,381 @@
...
+# ``0.0`` seconds after ``start_time`` and sampled every second:
 
 data = list(range(100, 200, 10))
-timestamps = list(range(10))
-test_ts = TimeSeries(name='test_timeseries', data=data, unit='m', timestamps=timestamps)
+time_series_with_rate = TimeSeries(name='test_timeseries',
+                                   data=data,
+                                   unit='m',
+                                   starting_time=0.0,
+                                   rate=1.0)
```
Comment (bendichter) [inline docs/gallery/general/file.py]:
```suggestion
time_series_with_rate = TimeSeries(
    name='test_timeseries',
    data=data,
    unit='m',
    starting_time=0.0,
    rate=1.0,
)
```

**Comment by bendichter** [comment]
@rly can we enable this feature? https://docs.readthedocs.io/en/stable/pull-requests.html I don't have the readthedocs credentials (or don't know them)

**Comment by rly** [comment]
Please review: https://pynwb--1403.org.readthedocs.build/en/1403/tutorials/general/file.html#sphx-glr-tutorials-general-file-py

**Comment by bendichter** [comment]
@rly I made a few more adjustments. Good to go from my end!

File: requirements-doc.txt [hunk 4]
```diff
@@ -3,6 +3,8 @@ sphinx
 matplotlib
 sphinx_rtd_theme
 sphinx-gallery
-allensdk>=2.11.0  # python 3.8 is not supported in allensdk<2.11
+h5py
+pynwb
+#allensdk>=2.11.0  # python 3.8 is not supported in allensdk<2.11
```
Comment (rly) [inline requirements-doc.txt]: why comment out allensdk? It is needed for the allensdk tutorial which is used in https://pynwb.readthedocs.io/en/stable/tutorials/domain/brain_observatory.html#sphx-glr-tutorials-domain-brain-observatory-py

Comment (bendichter) [reply requirements-doc.txt]: oh, yeah, I need to do this locally to get the docs to build. We can uncomment for the PR though

Comment (bendichter) [inline requirements-doc.txt]:
```suggestion
allensdk>=2.11.0  # python 3.8 is not supported in allensdk<2.11
```

**Review by rly — APPROVED** [review]

**Pull request merged**